submariner/fine-grained-connectivity.md

```diff
@@ -0,0 +1,86 @@
+# Fine Grained Connectivity
+
+## Summary
+
+Submariner currently provisions full mesh topology by default between the participating clusters.
+For some use-cases, we need fine-grained connectivity options. This proposal covers only the topology
+aspect of connectivity.
+
+This proposal aims to address the below use-case
+
+Hub-Spoke Topology -  Here only the client can talk to the server but not to the other clients to
+which the server is connected to.
+
+To achieve each hub and spoke will be grouped together into an isolated group of clusters called Clusterset.
+There will be a full mesh topology within the Clusterset, though in this use case it will be just two clusters
+in the Clusterset. A cluster can be a part of multiple Clusterset. So the hub can be a part of multiple
+Clustersets, where it connects with a spoke in each Clusterset
```

Comment:
Can a Spoke act like a Hub (in a different ClusterSet) for additional Spokes in this architecture?

Reply:
As mentioned elsewhere, a cluster being part of multiple Clustersets is going to be a sticking point with the u/s proposal. Perhaps we should not try to align with that for this use case and think in terms of a "_Broker domain_" (ie a broker namespace).  Maybe a cluster can only be part of one Clusterset but it can be part of multiple Broker domains. I think if we focus on the latter then we can avoid Clusterset, at least for now, since it's just an initial KEP right now.

Reply:
Yes spoke can be a hub too. Submariner does not differentiate between hub and spoke. It is based on the topology user creates certain clusters will be hub.